app/controllers/character_weapons_controller.rb

```diff
@@ -0,0 +1,74 @@
+class CharacterWeaponsController < ApplicationController
+  before_action :authenticate_user!
+  before_action :set_character_weapon, only: [:destroy, :show, :edit, :update]
+  before_action :verify_character_weapon, only: [:destroy, :show, :edit, :update]
+
+  def destroy
+    weapon_name = @character_weapon.weapon.name
+    @character_weapon.destroy
+    redirect_to @character_weapon.character, flash: { notice: "Weapon #{weapon_name} has been destroyed" }
+  end
+
+  def show
+    @weapon = @character_weapon.weapon
+  end
+
+  def new
+    @weapons = Weapon.all.order(rarity: :desc, price: :desc)
+    @character = current_character
+  end
+
+  def create
+    @character_weapon = CharacterWeapon.new(new_character_weapon_params)
```

Comment:
This is a bit weird, why do you need a `create` action if all params set on `new_character_weapon_params` are static? 🤔